test/unit/kernel/src/KernelThrow.cpp

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2022  René Widera, Mehmet Yusufoglu, Andrea Bocci
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#include <alpaka/core/RuntimeMacros.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+#include <alpaka/test/queue/Queue.hpp>
+
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+class KernelWithThrow
+{
+public:
+    template<typename TAcc>
+    ALPAKA_FN_ACC auto operator()(TAcc const& acc) const -> void
+    {
+        using Idx = alpaka::Idx<TAcc>;
+        using Dim = alpaka::Dim<TAcc>;
+        using Vec = alpaka::Vec<Dim, Idx>;
+        Vec const globalThreadIdx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+        Vec const globalThreadExtent = alpaka::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc);
+
+        // Map the three dimensional thread index into a one dimensional thread index space.
+        auto const linearizedGlobalThreadIdx = alpaka::mapIdx<1u>(globalThreadIdx, globalThreadExtent);
+
+        if(static_cast<unsigned>(linearizedGlobalThreadIdx[0u]) == 0u)
+        {
+            printf("linearizedGlobalThreadIdx: %u\n", linearizedGlobalThreadIdx[0]);
+            // Throw abort or std::runtime_error depending on acc type
+            ALPAKA_DEVICE_THROW("Exception thrown by the kernel");
+        }
+        alpaka::syncBlockThreads(acc);
+    }
+};
+
+template<typename T, typename Acc>
+void checkThrow()
+{
+    if constexpr(alpaka::accMatchesTags<Acc, T>)
+    {
```

Comment:
Why the extra complexity of using a `Tag` ?

Why not simply
```suggestion
template<typename Acc>
void checkThrow()
{
```
?

Reply:
Because `checkThrow` is called seperately for each enabled Acc.  Because it is called in a block by a TestAccs parameter type (tuple of Accs) which includes all enabled types.

`using TestAccs = alpaka::test::EnabledAccs<alpaka::DimInt<1u>, std::uint32_t>;

TEMPLATE_LIST_TEST_CASE("ThrowForCpuThreadAndSerial", "[kernel]", TestAccs)
{
    using Acc = TestType;
    // Test runtime-error exceptions. Aborts can not be catched.
    checkThrow<alpaka::TagCpuThreads, Acc>();`